examples/rust/file_transfer/examples/receiver.rs

```diff
@@ -82,19 +79,15 @@ impl Worker for FileReception {
 
 #[ockam::node]
 async fn main(ctx: Context) -> Result<()> {
-    // Initialize the TCP Transport.
-    let tcp = TcpTransport::create(&ctx).await?;
-
-    // Create a Vault to safely store secret keys for Receiver.
-    let vault = Vault::create();
+    let node = node(ctx);
```

Comment:
I'm a little bit concerned with how many decisions are made inside this function that are not visible to a user, and even if you dig inside this function, it's not really obvious what those decisions are and what other options you had.
For example, any real-world use-case would not want to use a `Vault` without persistence. I even think that the current implementation should have looked like this `Vault::create_without_persistence()`. But now it's created for you somewhere under the hood, and it will be super hard for a user to realize that their keys are gone after a restart and what should be done to fix that.
Should we explore options to make the user make this decision explicitly?

Reply:
We can leave this discussion for later and merge as it is, if you wish

Reply:
What I am planning to do is to add some examples showing how to go from this simple form of example to more complex examples where the implementation can be selected by the user. 

I can also:

 - add comments on top of this instantiation, to make it clear that it is in-memory only (like we we doing with `Vault::create()`
 - add rust comments in `fn node` to explain how to build more complex implementations for production use cases

Reply:
done